jobs/e2e-test.groovy

```diff
@@ -157,215 +153,86 @@ DEPLOYER_USER = params.DEPLOYER_USERNAME.replace("@", "")
           "gcloud", "--project", "khan-academy",
           "secrets", "versions", "access", "latest",
           "--secret", "Slack_api_token_for_slack_owl",
-      ]);
-
-      dir('webapp/testing/e2e') {
-         withEnv(["SLACK_TOKEN=${slackToken}"]) {
-            // notify-e2e-results returns a non-zero rc if it detects
-            // test failures. We set the job to UNSTABLE in that case
-            // (since we don't consider smoketest failures to be
-            // blocking).  But we still keep on running the rest of
-            // this script!
-            catchError(buildResult: "UNSTABLE", stageResult: "UNSTABLE",
-                       message: "There were test failures!") {
-               exec(notifyResultsArgs);
-            }
+   ]);
+}
 
-            // Let notify() know not to send any messages to slack, because we
-            // just did it here.
-            env.SENT_TO_SLACK = '1';
-         }
+def withGitHubToken(body) {
+   wrap([$class: 'MaskPasswordsBuildWrapper', varPasswordPairs: [[password: GITHUB_TOKEN]]]) {
+      withEnv(["GITHUB_TOKEN=${GITHUB_TOKEN}"]) {
+         body()
       }
    }
 }
 
+def _pullWebapp() {
+   // need script to start the workflow
+   kaGit.safeSyncToOrigin("git@github.com:Khan/webapp", GIT_SHA1, [], force=true);
+}
+
 // Determines if we are running the first or second smoke test.
-E2E_RUN_TYPE = (E2E_URL == "https://www.khanacademy.org" ? "second-smoke-test" : "first-smoke-test");
+IS_PRODUCTION = (E2E_URL == "https://www.khanacademy.org");
+E2E_RUN_TYPE = IS_PRODUCTION ? "second-smoke-test" : "first-smoke-test";
+E2E_MODE = IS_PRODUCTION ? "production" : "non-default";
 
 onWorker(WORKER_TYPE, '5h') {     // timeout
    notify([slack: [channel: params.SLACK_CHANNEL,
-                   thread: params.SLACK_THREAD,
-                   sender: 'Testing Turtle',
-                   emoji: ':turtle:',
-                   when: ['FAILURE', 'UNSTABLE']],
-           buildmaster: [sha: params.GIT_REVISION,
-                         what: E2E_RUN_TYPE]]) {
+                  thread: params.SLACK_THREAD,
+                  sender: 'Testing Turtle',
+                  emoji: ':turtle:',
+                  when: ['FAILURE', 'UNSTABLE']],
+          buildmaster: [sha: params.GIT_REVISION,
+                        what: E2E_RUN_TYPE]]) {
       initializeGlobals();
-      stage("Generate skipped list") {
-         _setupWebapp();
-         dir("webapp/services/static") {
-            sh("pnpm cypress:clean");
-            exec(["./dev/cypress/e2e/tools/gen-skipped-e2e-tests.js", "${E2E_URL}"]);
-            stash includes: SKIPPED_E2E_FILENAME, name: SKIPPED_STASH_ID;
-         };
-      }
+      _pullWebapp();
       stage("Run e2e tests") {
-         withEnv([
-            "COMMIT_INFO_BRANCH=${SHORT_REVISION_DESCRIPTION}",
-            "KA_SKIP_GEN=1",
-         ]) {
-            runAllTestClients();
+         withGitHubToken {
+            withEnv(["SLACK_TOKEN=${SLACK_TOKEN}"]) {
```

Comment:
In this case, you could avoid having to read the secret via an envvar by having notify-workflow-status.ts load the secret itself.  (Either using the js gsm library, or just calling out to gcloud in a subprocess.) 
 Then that script won't depend on envvars as much either, which is a maintainability win.